src/ConnectionRequest.jl

```diff
@@ -8,6 +8,43 @@ using ..ConnectionPool
 using MbedTLS: SSLContext
 import ..@debug, ..DEBUG_LEVEL
 
+const HTTP_PROXY = Ref{String}()
+const HTTPS_PROXY = Ref{String}()
+const NO_PROXY = Set{String}()
+
+gethttp_proxy() = isdefined(HTTP_PROXY, :x) ? HTTP_PROXY[] : nothing
+gethttps_proxy() = isdefined(HTTPS_PROXY, :x) ? HTTPS_PROXY[] : nothing
+
+getproxy(scheme) = scheme == "http" ? gethttp_proxy() : scheme == "https" ? gethttps_proxy() : nothing
```

Comment:
Why not read the `ENV` var in these functions?
For a long running REPL I can imagine people wanting to change proxy settings  without restarting.

And the code would be simpler

Reply:
This will get called for ***every*** request, so I'm a little worried about performance, but maybe that wouldn't cost too much.

Reply:
```
julia> @btime (()->ENV["PATH"])()
  39.791 ns (1 allocation: 144 bytes)
```

```
julia> const Path = ENV["PATH"];

julia> @btime (()->Path)()
  1.487 ns (0 allocations: 0 bytes)
```


I am leaning towards it being negliabale compared to the IO time.

Reply:
Ok, I've refactored it a bit.